include/nanobind/nb_cast.h

```diff
@@ -286,7 +286,8 @@ template <> struct type_caster<char> {
         value = PyUnicode_AsUTF8AndSize(src.ptr(), &size);
         if (!value) {
             PyErr_Clear();
-            return false;
+            // optimize for src being a string, check for None afterwards
+            return src.is_none();
```

Comment:
Maybe you want:
```
    return src.is_none() && flags & (uint8_t) cast_flags::accepts_none
```
where `flags` is the name given to the second argument to `from_python()` above.
But I'm not the expert on such things....

Reply:
I think it's not needed. `accepts_none` is checked in `nb_func_vectorcall_complex()` and None can't go through `nb_func_vectorcall_simple()`. I remember testing it, but I could be missing something.